### ISSUE TITLE
title_format not working at all

### ISSUE THREAD

**Issue by Ojd2000** [swaywm/sway#6688]
**Sway Version:**
1.6.1-2

**Note**
I tried, as suggested by the bug form, to run `sway -d` in my tty2 with the default config file, but sway was completely unresponsive, no key was working.
I currently do not have any more time for debugging; I will run some when I find some spare time. Sorry.

**Description:**
I have a few `for_window` clauses which I use to make some of my terminal windows floating, sticky, and resize them. I use [wezterm](https://github.com/wez/wezterm) as my terminal emulator. They have always worked.
Some days ago I added to them a `title_format` option to change those windows' titles, both with a "fixed string", and with a string that uses `%title` placeholder, but I couldn't come up with a way to make them work.
I did not edit anything else, and sway did not report any syntax error. Having read the #6606 issue, I tried removing the `floating` option to see if it could be related, but it did not help.

**Comment by orgarten** [comment]
Where do you want the title to be change?

I kind of have the same problem. The title is supposed to be show in waybar but apparently that is a property called `name`.

If you enable borders, you can see the changed title in the top border:
![image](https://user-images.githubusercontent.com/10799869/144449034-dbf09814-ba92-4764-931e-252d832dd235.png)

Not sure if this is helpful at all in figuring out whats going on.

**Comment by Ojd2000** [comment]
@orgarten Sorry for the late response.
Yes, I want it to be changed in waybar, but apparently `title_format` does modifications internal to sway only, as far as I understand (the modified title is not accessible from outside sway, e.g. with `swaymsg`, from what I see).
I think an option with this title to `swamsg -t get_tree`'s output would be really useful in this case, then it would be up to other programs to use it (in this case, waybar).

**Comment by lethedata** [comment]
Yeah, the `title_format` option "Sets the format of window titles"[^1] which is that blue bar shown in @orgarten picture. Waybar pulls the title, app_id, shell, and marks as seen in `swaymsg -t get_tree` output.

For waybar, you can use the sway-window-module rewrite config to handle this:

```jsonc
"sway/window": {
	"format": "{shell} {title}",
	"rewrite": {
		"xdg_shell (.*)": "$1", // don't print "xdg_shell "
		"xwayland (.*)" : "[XWayland] $1" // formatted xwayland
	}
}
```

This is the equivalent window title output modified by `title_format` rule: 

```
for_window [shell="xwayland"] title_format "[XWayland] %title"
```

[^1]: From `sway(5)`. "Window titles" is a specific app term
